### ISSUE TITLE
Emigration No Longer Works (Dwarves Return)

### ISSUE THREAD

**Issue by ElsaTheHobo** [DFHack/dfhack#1409]
Dwarf Fortress Version: 44.12 x64
DFHack Version: 44.12-r1 x64

Expected outcome:
Dwarves leaving with `emigration enable` leave permanently

Actual outcome:
Sad dwarves leave, but return a few days later with an announcement popup `[Dwarf Name] has returned!`

It looks like a problem with the new raids feature, in that the game now checks for off-map units that should be on-map, and returns them. The nice announcement helps me at least expel them manually, which is nice. I was thinking, maybe emigration should use the new expel feature, instead of whatever it's doing now? The only issue is that it doesn't work on units with family that permanently live off-site (acknowledged bug, [http://www.bay12games.com/dwarves/mantisbt/view.php?id=10848](url)

Thanks for the DFhack!

**Comment by KCIV** [comment]
To add to this, the Exile script now does the same thing and no longer works.

Dwarfs leave and immediately return with either "with stories" or just 'returned'.

**Comment by lethosor** [comment]
I don't think that's one of our scripts, is it? I'm not seeing it under https://github.com/DFHack/scripts?files=1

**Comment by KCIV** [comment]
> I don't think that's one of our scripts, is it? I'm not seeing it under https://github.com/DFHack/scripts?files=1

You are correct, Huh. Its listed here http://dwarffortresswiki.org/index.php/Utility:DFHack/scripts and many other places, must be a different repository that was included in my lazynewbpack? Its listed under LS of DFHack for me as well.

Apologies.

**Comment by lethosor** [comment]
Yeah, there are some third party scripts that get distributed in packs.

**Comment by lethosor** [comment]
Yeah, there are some third party scripts that get distributed in packs.

**Comment by ab9rf** [comment]
I would like to note that this script (https://github.com/DFHack/scripts/blob/master/emigration.lua) was added to the script repository in _2015_, and (still today) works in the (unexpected) manner reported in this bug report.

**Comment by lethosor** [comment]
Yep - `exile` is unrelated, as far as I know, but the issue with `emigration` still stands.
